subsys/bluetooth/host/conn.c

```diff
@@ -2444,40 +2467,22 @@ bt_security_t bt_conn_get_security(const struct bt_conn *conn)
 
 void bt_conn_cb_register(struct bt_conn_cb *cb)
 {
-	cb->_next = callback_list;
-	callback_list = cb;
+	if (!sys_slist_find(&conn_cbs, &cb->_node, NULL)) {
+		sys_slist_append(&conn_cbs, &cb->_node);
+	}
```

Comment:
What do you think about adding an error log here? I think double registration is indicative of problems in the application code.

OTOH, I can see the convenience for the application. But then if it's not an error, we should explain this in the API documentation.

```suggestion
	} else {
		/* Note to users: `bt_disable` does not unregister these. */
		LOG_ERR("Can't register bt_conn_cb, already in use");
	}
```

Reply:
My opinion would be to add a return value to all register functions so that we can return `-EALREADY`, as we already do this with a significant subset of the register functions

Reply:
Do you consider this as an API change or not? I can create a separate commit for it.
See https://github.com/zephyrproject-rtos/zephyr/pull/73894#issuecomment-2157705707